lib/kubeclient/common.rb

```diff
@@ -66,6 +71,15 @@ def initialize_client(
       end
     end
 
+    def method_missing(method_sym, *args, &block)
+      if !@discovered && method_sym.to_s.match(/^(get|watch|delete|create|update|patch)_(.*)$/)
```

Comment:
> @moolitayer do we need to check for these or is it just an optimization that we could drop?

Since this is using a basic mechanism of the language I prefer to be as explicit as possible to avoid replacing any unexpected methods. I will extract the list as suggested.